packages/edit-site/src/components/list/actions/rename-menu-item.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+import { useEntityProp, store as coreStore } from '@wordpress/core-data';
+import { __ } from '@wordpress/i18n';
+import {
+	Button,
+	Flex,
+	FlexItem,
+	MenuItem,
+	Modal,
+	TextControl,
+} from '@wordpress/components';
+
+export default function RenameMenuItem( { template, onClose } ) {
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+
+	const { saveEditedEntityRecord } = useDispatch( coreStore );
+	const [ title, setTitle ] = useEntityProp(
+		'postType',
+		template.type,
+		'title',
+		template.id
+	);
+
+	async function onTemplateRename( event ) {
+		event.preventDefault();
+		setIsModalOpen( false );
+		onClose();
+
+		// Persist edited entity.
+		await saveEditedEntityRecord( 'postType', template.type, template.id );
+	}
+
+	return (
+		<>
+			<MenuItem
+				onClick={ () => {
+					setIsModalOpen( true );
```

Comment:
Why not calling `onClose` here once the modal is opened?

Reply:
Calling `onClose` here will unmount dropdown, so modal is never rendered.

Reply:
I see. Then why aren't we rendering the modal outside of the dropdown?

Reply:
No strong preference for this. I just thought having a separate component would be a little cleaner. Happy to move logic in the main file. Probably going to save a few lines of code as well :)

Reply:
BTW, I don't think we should be closing the modal for accessibility reasons. Closing the modal should move focus back to the Rename button again.

Reply:
I also like that behavior but wasn't sure if it was accessibility improvement.